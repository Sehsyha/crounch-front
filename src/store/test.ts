import { ActionTree, MutationTree, GetterTree, Module, ActionContext } from 'vuex';
import { RootState } from '.';

export type ActionCall<S, R> = (injectee: ActionContext<S, R>, payload?: any) => any;

export function createStoreModuleMock<S, A, M, G>(initialState: S, actionKeys: A, mutationKeys: M, getterKeys: G): Module<S, RootState> {
  const actions: ActionTree<S, RootState> = {};
  const mutations: MutationTree<S> = {};
  const getters: GetterTree<S, RootState> = {};

  Object.keys(actionKeys).forEach(key => actions[key] = jest.fn());
  Object.keys(mutationKeys).forEach(key => mutations[key] = jest.fn());
  Object.keys(getterKeys).forEach(key => getters[key] = jest.fn());

  return {
    namespaced: true,
    actions,
    getters,
    mutations,
    state: initialState,
  }
}
