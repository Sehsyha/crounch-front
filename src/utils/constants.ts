export const TOKEN_STORAGE_KEY = 'token';

export const DEFAULT_CATEGORY_ID = '00000000-0000-0000-0000-000000000000'
export const DEFAULT_CATEGORY_NAME = 'Divers'
