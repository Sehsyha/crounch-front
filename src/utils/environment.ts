export function getAPIURL(): string {
  return process.env.NODE_ENV === 'production' ? 'http://crounch.me:3000/' : 'http://localhost:3000/';
}
