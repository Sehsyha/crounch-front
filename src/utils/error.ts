export interface ErrorBody {
  code: string;
  description: string;
}

export class FetchError extends Error {
  status: number;
  code: string;
  description: string;

  constructor(message: string, status: number, body: ErrorBody) {
    super(message)
    this.status = status
    this.code = body.code
    this.description = body.description
  }
}
