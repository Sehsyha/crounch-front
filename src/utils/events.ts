export enum Events {
  ADD_PRODUCT = 'add-product',
  DELETE_PRODUCT = 'delete-product'
}
