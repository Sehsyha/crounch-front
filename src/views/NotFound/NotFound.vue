<template>
  <div>
    <h1>Page non trouv√©e &#x1F632;</h1>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
export default class NotFound extends Vue { }
</script>
