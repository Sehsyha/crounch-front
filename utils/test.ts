import { shallowMount } from '@vue/test-utils';

export function shallowComponent(component: any, values: {}) {
  return shallowMount(component, {
    data() {
      return {
        ...values
      }
    }
  });
}
