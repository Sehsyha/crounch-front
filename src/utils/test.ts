import Vue from 'vue';
import VueRouter from 'vue-router';

import Vuex, { ActionContext, ActionTree, ModuleTree } from 'vuex';
import { shallowMount, VueClass, createLocalVue, Slots } from '@vue/test-utils';

import { RootState, ActionCall } from '@/store';

export interface ShallowOptions {
  values?: object;
  modules?: ModuleTree<RootState>;
  slots?: Slots;
}

const emptyOptions: ShallowOptions = {
  values: {},
  modules: {},
  slots: {}
};

export function shallowComponent<V extends Vue>(component: VueClass<V>, { values, modules, slots }: ShallowOptions = emptyOptions) {
  const localVue = createLocalVue();

  localVue.use(Vuex);

  const router = new VueRouter();

  const store = new Vuex.Store<RootState>({
    modules,
  });

  return shallowMount(component, {
    attachToDocument: true,
    propsData: {
      ...values,
    },
    slots,
    store,
    router,
    localVue,
    stubs: ['router-link', 'router-view'],
    mocks: {
      $router: {
        push: jest.fn()
      }
    }
  });
}

export const callAction = <S>(context: ActionContext<S, RootState>, actions: ActionTree<S, RootState>) => (
  key: string,
  params?: any
): any => {
  return (actions[key] as ActionCall<S, RootState>)(context, params);
};

export function initContext<S>(context: Partial<ActionContext<S, RootState>>): ActionContext<S, RootState> {
  const emptyContext = createEmptyActionContext<S>();

  return {
    ...emptyContext,
    ...context,
  };
}

function createEmptyActionContext<S>(): ActionContext<S, RootState> {
  return {
    state: {},
    getters: {},
    dispatch: {},
    commit: {},
    rootState: {
      dummy: '',
    },
    rootGetters: {},
  } as ActionContext<S, RootState>;
}
