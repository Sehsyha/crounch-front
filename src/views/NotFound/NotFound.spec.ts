import { shallowComponent } from '@/utils/test';
import NotFound from './NotFound.vue';


describe('NotFound', () => {
  it('Should render a not found page.', () => {
    const wrapper = shallowComponent(NotFound);

    expect(wrapper).toMatchSnapshot();
  });
});
