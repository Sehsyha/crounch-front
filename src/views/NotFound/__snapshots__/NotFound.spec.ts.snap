// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Should render a not found page. 1`] = `
<div>
  <h1>Page non trouvÃ©e ðŸ˜²</h1>
</div>
`;
